proposals/p3848.md

```diff
@@ -0,0 +1,576 @@
+  thread.Join();
+}
+```
+
+### Default Capture Mode
+
+**Proposal**: By default, there is no capturing in functions. The lack of any
+square brackets is the same as an empty pair of square brackets. Users can opt
+into capturing behavior. This is done either by way of individual explicit
+captures, or more succinctly by way of a default capture mode. The default
+capture mode mirrors the syntax `[=]` and `[&]` capture modes from C++ (written
+in Carbon as `[copy]` and `[ref]`) by being the first thing to appear in the
+square brackets.
+
+```
+fn Foo() {
+  let handle: Handle = Handle.Get();
+  fn MyThread[copy]() {
+    handle.Process();  // `handle` is copy captured due to the default capture
+                       // mode specifier of `copy`
+  }
+  var thread: Thread = Thread.Make(MyThread);
+  thread.Join();
+}
+```
+
+**Alternative Considered**: Previously, it was proposed that the default capture
+mode would come after all the explicit captures. In addition, it was proposed
+that the lack of any square brackets opted the function into capturing by
+default with an implicitly defined default capture mode. These behaviors do not
+mirror lambdas in C++ and so were decided against. Primarily, it was recognized
+that it's valuable to be able to intermix explicit captures with deduced
+parameters and fields in any order that makes the most sense for the context.
+Without a clear justification for a rule that says they can't intermix, the more
+flexible behavior was favored.
+
+### Capture Modes
+
+**Proposal**: `copy` and `ref` are the two primary forms of function captures.
+They each have companion `const` versions. They are syntactically distinct from
+`let` and `var`. `let` and `var` can appear in binding patterns, such as
+[function fields](#function-fields), while `copy`, `ref`, `const copy` and
+`const ref` can appear as function captures. The two sets of keywords cannot
+appear in the same syntactic positions.
+
+| Capture Mode Syntax |                     Corresponding Semantics                      |
+| :-----------------: | :--------------------------------------------------------------: |
+|       `copy`        |     Capture "by-copy" where the resulting capture is mutable     |
+|    `const copy`     | Capture "by-const-copy" where the resulting capture is immutable |
+|        `ref`        |        Capture "by-reference" behaving as a C++ reference        |
+|     `const ref`     |  Capture "by-const-reference" behaving as a C++ const reference  |
+
+Capture modes can be used as
+[default capture mode specifiers](#default-capture-mode) or for explicit
+captures as shown in the example code below.
+
+```
+fn Example {
+  var by_copy: i32 = 0;
+  var by_const_copy: i32 = 0;
+  var by_reference: i32 = 0;
+  var by_const_reference: i32 = 0;
+
+  let lambda: auto = fn [copy by_copy,
+                         const copy by_const_copy,
+                         ref by_reference,
+                         const ref by_const_reference] {
```

Comment:
I'm having a hard time getting my brain to distinguish the keywords from the identifiers in this example. I can't tell if that's a problem with the syntax itself, or an artifact of the fact that these identifiers all have the keywords embedded in them. Could you rename the variables to avoid the latter problem?